ansible/roles/prometheus-server/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+---
+
+- name: ensure {{ prometheus_group }} group exists
+  become: yes
+  group:
+    name: "{{ prometheus_group }}"
+    system: yes
+
+- name: ensure {{ prometheus_user }} user exists
+  become: yes
+  user:
+    name: "{{ prometheus_user }}"
+    group: "{{ prometheus_user }}"
+    system: yes
+
+- name: check existing volume mounts
+  set_fact:
+    volume_mounted: "{{ ansible_mounts | selectattr('mount', 'match', prometheus_data_dir) | list | length > 0 }}"
+
+- name: configure volume mounts
+  include_role:
+    name: mounted-volume
+  when:
+    - not volume_mounted
+    - volume_mounts is defined
+    - volume_mounts != ""
```

Comment:
This is potentially over-engineered from the perspective of building an AMI—the volume mount is guaranteed to not exist at this point as no provisioning to mount the volume has been performed, and given that the role will never be run more than once against the same instance the `not volume_mounted` condition in combination with the previous `set_fact` task are redundant here. As for the remaining conditions, `volume_mounts` is defined in the host variables file making this check similarly redundant (if volume mounts are removed at a later time then the `mounted-volume` role itself becomes redundant and should similarly be removed at the same point).

Instead I would suggest removing both the task and the previous `set_fact` task in favour of adding the role directly to the play:

```yml
- hosts: prometheus
  become: true
  roles:
    - mounted-volume
    - prometheus-server
```

This also makes the provisioning more consistent with [other AMI builds]([concourse-ami](https://github.com/companieshouse/smartpan-ami/blob/master/ansible/playbook.yml#L9)).

Reply:
Yeah, I'd left this in since chatting with Jimmy re the tarball logic (which you spotted later too). But I'll remove it - it's a fair shout.

Reply:
PS - I've not moved the `mounted-volume` role to the playbook since it needs the user and group to exist (and that happens in the `prometheus-server` role). I guess I could move `mounted-volume` role to follow `prometheus-server`. I think that should work, but it doesn't seem a logical order.
See what you think.

Reply:
As per convo, moved the mounted-volume to the end of the playbook.